metadata-ingestion/src/datahub/ingestion/source/schema_inference/csv_tsv.py

```diff
@@ -0,0 +1,74 @@
+from typing import IO, Dict, List, Type
+
+from tableschema import Table
+
+from datahub.ingestion.source.schema_inference.base import SchemaInferenceBase
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    ArrayTypeClass,
+    BooleanTypeClass,
+    DateTypeClass,
+    NullTypeClass,
+    NumberTypeClass,
+    RecordTypeClass,
+    SchemaField,
+    SchemaFieldDataType,
+    StringTypeClass,
+    TimeTypeClass,
+    UnionTypeClass,
+)
+
+# see https://github.com/frictionlessdata/tableschema-py/blob/main/tableschema/schema.py#L545
+tableschema_type_map: Dict[str, Type] = {
+    "duration": TimeTypeClass,
+    "geojson": RecordTypeClass,
+    "geopoint": RecordTypeClass,
+    "object": RecordTypeClass,
+    "array": ArrayTypeClass,
+    "datetime": TimeTypeClass,
+    "time": TimeTypeClass,
+    "date": DateTypeClass,
+    "integer": NumberTypeClass,
+    "number": NumberTypeClass,
+    "boolean": BooleanTypeClass,
+    "string": StringTypeClass,
+    "any": UnionTypeClass,
+}
+
+MAX_ROWS = 100
```

Comment:
What is the purpose of this and how did we arrive at this value of 100? Should this be made configurable?

Reply:
Good point — I've made it configurable now.